main.c

```diff
@@ -0,0 +1,721 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+// Define the maximum number of schools, layers, and classes
+#define MAX_LAYERS 12
+#define MAX_CLASSES 10
+
+enum menu_inputs {
+    Insert = '0',
+    Delete = '1',
+    Edit = '2',
+    Search = '3',
+    Showall = '4',
+    Top10 = '5',
+    UnderperformedStudents = '6',
+    Average = '7',
+    Export = '8',
+    Exit = '9'
+};
+
+// Structure to store student details
+struct Student {
+    char firstName[50];
+    char lastName[50];
+    char phoneNumber[20];
+    int gradeNumber;
+    int classNumber;
+    int grades[10];
+    struct Student* next;
+};
+
+struct TopTenStudents {
+    struct Student* topTenStudentsOfLayers[12][10];
+};
+
+// Structure to store class details
+struct Class {
+    struct Student* students;
+    int studentCount;
+};
+
+// Structure to store layer details
+struct Layer {
+    struct Class classes[MAX_CLASSES];
+};
+
+// Structure to store school details
+struct School {
+    struct Layer layers[MAX_LAYERS];
+};
+
+// Function to create a new student node
+struct Student* createStudentNode(const char* firstName, const char* lastName, const char* phoneNumber, int gradeNumber, int classNumber, int* grades) {
+    struct Student* newStudent = (struct Student*)malloc(sizeof(struct Student));
+    strcpy(newStudent->firstName, firstName);
+    strcpy(newStudent->lastName, lastName);
+    strcpy(newStudent->phoneNumber, phoneNumber);
+    newStudent->gradeNumber = gradeNumber;
+    newStudent->classNumber = classNumber;
+    memcpy(newStudent->grades, grades, sizeof(int) * 10);
+    newStudent->next = NULL;
+    return newStudent;
+}
+
+// Function to add a student to the class using a linked list
+void addStudentToClass(struct Class* classPtr, struct Student* student) {
+    if (classPtr->students == NULL) {
+        classPtr->students = student;
+    } else {
+        struct Student* current = classPtr->students;
+        while (current->next != NULL) {
+            current = current->next;
+        }
+        current->next = student;
```

Comment:
If the linked list has 10 million nodes, will you still go all the way to the end to add a new student?
Think of an efficient way to add a student to a linked list.

Reply:
Fixed it, I add new student to beginning of list.